### PR TITLE
ASP.NET Core Module compatibility

### LINKED ISSUE #21910: ASP.NET Core Module compatibility
_This question might be more related to the actual library rather than the documentation, but the answer should probably be available in the documentation, so I'm creating the issue here._

There is a new version of the ASP.NET Core Module v2 binaries made available with every .NET release. Does this mean that the only supported way of running ASP.NET applications in IIS is via the version of the ASP.NET Core Module released with a particular .NET release?

Some smoke testing indicates that it is possible to run e.g. NET 5.0 apps with the hosting bundle released with 3.1. This is in line with what is mentioned here: https://docs.microsoft.com/en-us/answers/questions/176303/net-core-31-apps-with-the-net-5-hosting-bundle-are.html

There are two major reasons this compatibility is important:

1. Being able to host multiple applications in IIS on the same server, but using different versions of .NET/.NET Core.
2. Avoid having to restart IIS and take down other applications when installing an application/service using a newer .NET version.

Both these points are important for installations where different services/applications have their own release cycle, and making other services unavailable during the upgrade of one service is highly undesirable. It can be especially problematic forcing IIS or server restarts for on-premise installations where the application developer is not in control of the hosting environment. 

So I'm hoping for an answer that the ASP.NET Core Module v2 is compatible with multiple versions. It would great if e.g. the ASP.NET Core Module released with 3.1 could host applications using .NET 6. No matter the answer, I suggest adding that information to the documentation.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fe69b3a9-896f-e189-0c2c-f87ff735df9c
* Version Independent ID: 97c8549b-66ad-0a23-1971-784fb923042a
* Content: [ASP.NET Core Module](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-5.0)
* Content Source: [aspnetcore/host-and-deploy/aspnet-core-module.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/host-and-deploy/aspnet-core-module.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #21910

[Internal Review Topic (links to section in the 5.0 content)](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-2.1&branch=pr-en-us-21914#install-aspnet-core-module)

Thanks @avendel! :rocket:

No cross-link to *Announcements* here. I'll note here in passing that there is no main *to keep up with Announcements, see ...* cross-link outside of mentions in *Migration* topics (cc: @Rick-Anderson on that in case it should be called out in the [ASP.NET Core Introduction article](https://docs.microsoft.com/aspnet/core/introduction-to-aspnet-core) ... I can add it to this PR if you want ... it's just a one-liner or an *Additional resources* cross-link there I think 🤔).

### CODE REVIEW

**Review by jkotalik — APPROVED** [review]

File: aspnetcore/host-and-deploy/aspnet-core-module.md [hunk 1]
```diff
@@ -25,11 +25,13 @@ There are trade-offs between each of the hosting models. By default, the in-proc
 
 ## Install ASP.NET Core Module
 
+The ASP.NET Core Module is installed with the .NET Core Runtime from the [.NET Core Hosting Bundle](xref:host-and-deploy/iis/hosting-bundle). The ASP.NET Core Module is forward and backward compatible with releases of .NET.
```
Comment (jkotalik) [inline aspnetcore/host-and-deploy/aspnet-core-module.md:28]: I'd maybe mention LTS here.

**Comment by guardrex** [comment]
Like that :point_up:? Look good?

btw - I was thinking that such a link for these three spots would go something like this if we did it ...

> Breaking changes or security advisories that apply to the ASP.NET Core Module or .NET Core Hosting Bundle are reported on the \[ASP.NET Core Announcements repo](https://github.com/aspnet/Announcements/issues). Annoucements can be filtered with \`label:{VERSION}\`, where the \`{VERSION}\` placeholder is a .NET release version (for example \`6.0.0\`).

Let me know if you want that. Otherwise, I'll just wait to see if Rick wants the Introduction to have something on that repo.

**Comment by jkotalik** [comment]
I think the first is fine. And LGTM.

File: aspnetcore/host-and-deploy/aspnet-core-module.md [hunk 2]
```diff
@@ -25,11 +25,15 @@ There are trade-offs between each of the hosting models. By default, the in-proc
 
 ## Install ASP.NET Core Module
 
+The ASP.NET Core Module is installed with the .NET Core Runtime from the [.NET Core Hosting Bundle](xref:host-and-deploy/iis/hosting-bundle). The ASP.NET Core Module is forward and backward compatible with LTS releases of .NET.
+
+Breaking changes or security advisories that apply to the ASP.NET Core Module or .NET Core Hosting Bundle are reported on the [Announcements repo](https://github.com/aspnet/Announcements/issues). Announcements can be filtered with `label:{VERSION}`, where the `{VERSION}` placeholder is a .NET release version (for example, `label:6.0.0`).
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/aspnet-core-module.md:30]: Only a suggestion. Filtered made me look in the **Filters**. Mine is probably too verbose, folks can figure it out.

```suggestion
Breaking changes or security advisories that apply to the ASP.NET Core Module or .NET Core Hosting Bundle are reported on the [Announcements repo](https://github.com/aspnet/Announcements/issues). Announcements displayed can be limited to a specific version the with `label:{VERSION}`, where the `{VERSION}` placeholder is a .NET release version. For example, selecting **6.0.0** from the **Label** fiters displays only issues for the 6.0.0 release.
```

File: aspnetcore/host-and-deploy/aspnet-core-module.md [hunk 3]
```diff
@@ -137,7 +141,9 @@ The ASP.NET Core Module can also:
 
 ## How to install and use the ASP.NET Core Module
 
-For instructions on how to install the ASP.NET Core Module, see [Install the .NET Core Hosting Bundle](xref:host-and-deploy/iis/index#install-the-net-core-hosting-bundle).
+For instructions on how to install the ASP.NET Core Module, see [Install the .NET Core Hosting Bundle](xref:host-and-deploy/iis/index#install-the-net-core-hosting-bundle). The ASP.NET Core Module is forward and backward compatible with LTS releases of .NET.
+
+Breaking changes or security advisories that apply to the ASP.NET Core Module or .NET Core Hosting Bundle are reported on the [Announcements repo](https://github.com/aspnet/Announcements/issues). Announcements can be filtered with `label:{VERSION}`, where the `{VERSION}` placeholder is a .NET release version (for example, `label:6.0.0`).
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/aspnet-core-module.md:146]: See previous suggestion.

**Review by Rick-Anderson — APPROVED** [review]
See #21915 - seems like this is the way to go for 7.0. Just update the include for 7.0, no need to make a duplicate include, 6.0 folks can see the 7.0 text.

**Comment by guardrex** [comment]
> See #21915 - seems like this is the way to go for 7.0. Just update the include for 7.0, no need to make a duplicate include, 6.0 folks can see the 7.0 text.

Except that that's too specific to be used across the whole repo. There's probably a way to make it more generic.

These two PRs are stepping on either other a bit ... 🐔 and 🥚 scenario. I recommend that we either kill the added bits here and work the whole thing from the other PR or kill the other PR and work the INCLUDES file in here.

File: aspnetcore/host-and-deploy/aspnet-core-module.md (hunk 2 above)
Comment (guardrex) [reply aspnetcore/host-and-deploy/aspnet-core-module.md:30]: Yes on the use of the DDL. That's a better way to go.

**Comment by Rick-Anderson** [comment]
> Except that that's too specific to be used across the whole repo. There's probably a way to make it more generic.
> 
> These two PRs are stepping on either other a bit ... 🐔 and 🥚 scenario. I recommend that we either kill the added bits here and work the whole thing from the other PR or kill the other PR and work the INCLUDES file in here.

Or just go with what you have.

File: aspnetcore/host-and-deploy/aspnet-core-module.md (hunk 2 above)
Comment (Rick-Anderson) [reply aspnetcore/host-and-deploy/aspnet-core-module.md:30]: S&M, include is over-kill

**Comment by guardrex** [comment]
I'll make it generic in an INCLUDES file here. I'm thinking like ...

> Breaking changes or security advisories are reported on the \[Announcements repo](https://github.com/aspnet/Announcements/issues). Announcements can be limited to a specific version by selecting a **Label** filter.

I shortened what you had. I think the readers would get it from this.

**Comment by Rick-Anderson** [comment]
Good idea, that's plenty.

**Comment by guardrex** [comment]
@Rick-Anderson ... Ok ... try it now :eyes:.

**Comment by guardrex** [comment]
btw ---- I **_WILL_** use that INCLUDE in one or a couple of Blazor docs. I think that's a good thing to remind on in a few critical spots.

**Pull request merged**